sdk/keyvault/keyvault-admin/src/accessControlModels.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as coreHttp from "@azure/core-http";
+import { SUPPORTED_API_VERSIONS } from "./constants";
+
+/**
+ * The optional parameters accepted by the Key Vault's AccessControlClient
+ */
+export interface AccessControlClientOptions extends coreHttp.PipelineOptions {
+  /**
+   * The accepted versions of the Key Vault's service API.
+   */
+  serviceVersion?: SUPPORTED_API_VERSIONS;
+}
+
+/**
+ * A Key Vault role assignment.
+ */
+export interface KeyVaultRoleAssignment {
+  /**
+   * The role assignment ID.
+   */
+  readonly id?: string;
+  /**
+   * The role assignment name.
+   */
+  readonly name?: string;
+  /**
+   * The role assignment type.
+   */
+  readonly type?: string;
+  /**
+   * Role assignment properties.
+   */
+  properties?: KeyVaultRoleAssignmentPropertiesWithScope;
+}
+
+/**
+ * VaA list of Key Vault permissions.
+ */
+export interface KeyVaultPermission {
+  /**
+   * Allowed actions.
+   */
+  actions?: string[];
+  /**
+   * Denied actions.
+   */
+  notActions?: string[];
+  /**
+   * Allowed Data actions.
+   */
+  dataActions?: string[];
+  /**
+   * Denied Data actions.
+   */
+  notDataActions?: string[];
+}
+
+/**
+ * A Key Vault role definition.
+ */
+export interface KeyVaultRoleDefinition {
+  /**
+   * The role definition ID.
+   */
+  readonly id?: string;
+  /**
+   * The role definition name.
+   */
+  readonly name?: string;
+  /**
+   * The role definition type.
+   */
+  readonly type?: string;
+  /**
+   * The role name.
+   */
+  roleName?: string;
+  /**
+   * The role definition description.
+   */
+  description?: string;
+  /**
+   * The role type.
+   */
+  roleType?: string;
+  /**
+   * A list of Key Vault permissions.
+   */
+  permissions?: KeyVaultPermission[];
+  /**
+   * Role definition assignable scopes.
+   */
+  assignableScopes?: string[];
+}
+
+/**
+ * Role assignment properties.
+ */
+export interface KeyVaultRoleAssignmentProperties {
+  /**
+   * The role definition ID used in the role assignment.
+   */
+  roleDefinitionId?: string;
```

Comment:
is this really optional?

Reply:
Not optional! I updated the generated files, and the swagger corrected this.

Reply:
After recent feedback from the .Net team, through Teams, we agreed on enforcing most of the properties. So, please review again!